libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationViewModel.kt

```diff
@@ -21,25 +26,45 @@ internal class DropInNavigationViewModel : ViewModel() {
     private val _navigationState = MutableStateFlow<NavigationState>(NavigationState.FreeDrive)
     val navigationState = _navigationState.asStateFlow()
 
+    @Suppress("PropertyName")
+    val _activeNavigationStarted = MutableStateFlow(false)
+    val activeNavigationStarted = _activeNavigationStarted.asStateFlow()
+
+    private val _onBackPressedEvent = MutableSharedFlow<Unit>(
+        replay = 0,
+        extraBufferCapacity = 1,
+        onBufferOverflow = BufferOverflow.DROP_OLDEST
+    )
+    val onBackPressedEvent = _onBackPressedEvent.asSharedFlow()
+
     /**
      * These classes are accessible through MapboxNavigationApp.getObserver(..)
      */
     val replayViewModel = ReplayViewModel()
     val audioGuidanceViewModel = MapboxAudioViewModel()
     val locationViewModel = LocationViewModel()
-    val routesViewModel = RoutesViewModel()
+    val routesViewModel = RoutesViewModel(
+        navigationState,
+        locationViewModel.state
+    )
     val cameraViewModel = CameraViewModel()
-    val navigationObservers = listOf(
+    private val navigationStateManager = NavigationStateManager(
+        routesViewModel.state.map { it.destination },
+        _navigationState,
+        routesViewModel.state.map { it.navigationStarted },
+    )
+    private val navigationObservers = listOf(
         replayViewModel,
         audioGuidanceViewModel,
         locationViewModel,
         routesViewModel,
         cameraViewModel,
+        navigationStateManager
         // TODO can add more mapbox navigation observers here
     )
 
-    fun updateState(state: NavigationState) {
-        _navigationState.value = state
+    fun onBackPressed() {
+        _onBackPressedEvent.tryEmit(Unit)
```

Comment:
This should also be moved to the BackPressedViewModel

Reply:
I leave BackPress handling to you. This is just a temp solution to get back press working.